src/main/ngapp/accio/src/app/img-gallery/img-gallery.component.ts

```diff
@@ -1,16 +1,115 @@
 import { Component, OnInit } from '@angular/core';
-import { imageUrls } from '../images'
+import { ActivatedRoute } from '@angular/router';
+import { FormGroup, FormControl } from '@angular/forms';
+import { PostBlobsService } from '../post-blobs.service';
+import * as $ from 'jquery';
 
 @Component({
   selector: 'app-img-gallery',
   templateUrl: './img-gallery.component.html',
   styleUrls: ['./img-gallery.component.css']
 })
 export class ImgGalleryComponent implements OnInit {
-  imageUrls = imageUrls;
-  constructor() { }
+  uploadImageForm: FormGroup;
+  formData: FormData;
+  mode: string = 'create';
+  projectId: string;
+
+  displayUpload: boolean = false;
+  displayImages: boolean = false;
+
+  // Holds images fetched from datastore
+  imageArray: Array<any>;
+
+  constructor(
+    private route: ActivatedRoute,
+    private postBlobsService: PostBlobsService,
+  ) { }
 
   ngOnInit(): void {
+    this.uploadImageForm = new FormGroup({
+      imgName: new FormControl(),
+      image: new FormControl(),
+      tags: new FormControl()
+    });
+
+    //creates the form data of parameters to be sent to servlet
+    this.formData = new FormData();
+
+    // Set the project id, first try if the 
+    //    project is a new project and there are query keys
+    this.route.queryParams.subscribe(params => {
+      this.projectId = params['proj-id'];
+    });
+    //if no keys then just get the url params
+    if (!this.projectId) {
+      this.route.paramMap.subscribe(params => {
+        this.projectId = params.get('proj-id');
+      });
+    }
+
+    console.log('projID ' + this.projectId);
+    console.log('now fetching...')
+
+    // Get the blobstore url initalized and show the form
+    this.postBlobsService.fetchBlob();
+    this.displayUpload = true;
+    this.getImages();
   }
 
+  async getImages(): Promise<any> {
+    console.log('fetching from projectID: ' + this.projectId);
+    let fetchUrl = '/blobs?' + $.param({
+      'proj-id': this.projectId
+      /* add ability to sort by:
+      *     tag
+      *     img-name
+      *     mask-name
+      *     with-masks	Boolean
+      *     sort-img	“asc” or “dsc”
+      *     sort-mask	"asc” or “dsc” 
+      */
+    });
+
+    // fetchUrl returns a list of image objects: 'url', 'name', 'utc', 'tags[]', 'masks[]'
+    const response = await fetch(fetchUrl);
+    const imageContent = await response.json();
+    this.imageArray = imageContent;
+
+    if (this.imageArray.length > 0) {
+      this.displayImages = true;
+    }
+  }
+
+  /** 
+   *  @param formData is initalized with values given by user.
```

Comment:
I'm not sure if '@param' is the correct semantic here. Might be a bit confusing since it's 'imgName', 'tags', and 'image' that are the actual arguments received from the form. But I'm also not sure what the correct wording is.